### PR TITLE
CEDS-1767 Removed old consolidation endpoints

### PR DESCRIPTION
**Pull request by PatrykRudnicki**

### CODE REVIEW

File: app/uk/gov/hmrc/exports/movements/services/SubmissionService.scala [hunk 1]
```diff
@@ -58,14 +58,31 @@ class SubmissionService @Inject()(
     }
 
   def submitConsolidation(eori: String, consolidation: Consolidation)(implicit hc: HeaderCarrier): Future[Unit] = {
-    val requestXml = wcoMapper.generateConsolidationXml(consolidation)
+    val requestXml = ileMapper.generateConsolidationXml(consolidation)
 
-    submitRequest(consolidation.buildSubmissionContext(eori, requestXml))
+    customsInventoryLinkingExportsConnector.sendInventoryLinkingRequest(eori, requestXml).flatMap {
+
+      case CustomsInventoryLinkingResponse(ACCEPTED, Some(conversationId)) =>
```
Comment (ed0906) [inline app/uk/gov/hmrc/exports/movements/services/SubmissionService.scala]: Can we put this in the connector layer? it doesn't seem like Service logic to me

Connector should just return `Future[String]` or `Future.failed`

Comment (bambuchaAdm) [reply app/uk/gov/hmrc/exports/movements/services/SubmissionService.scala]: Maybe we should move Accepted to class name level - then http logic would not leak to http - on the other hand - failied future from connetor always could be recoever

File: test/unit/uk/gov/hmrc/exports/movements/services/ILEMapperSpec.scala [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+package unit.uk.gov.hmrc.exports.movements.services
+
+import uk.gov.hmrc.exports.movements.models.consolidation.Consolidation
+import uk.gov.hmrc.exports.movements.models.consolidation.ConsolidationType.ASSOCIATE_DUCR
+import uk.gov.hmrc.exports.movements.models.submissions.ActionType.DucrAssociation
+import uk.gov.hmrc.exports.movements.services.ILEMapper
+import unit.uk.gov.hmrc.exports.movements.base.UnitSpec
+
+class ILEMapperSpec extends UnitSpec {
```
Comment (bambuchaAdm) [inline test/unit/uk/gov/hmrc/exports/movements/services/ILEMapperSpec.scala]: I don't know there full name would be better for saerching

File: app/uk/gov/hmrc/exports/movements/services/SubmissionService.scala (hunk 1 above)
Comment (PatrykRudnicki) [reply app/uk/gov/hmrc/exports/movements/services/SubmissionService.scala]: Probably I agree more with @SteveSugden so I'll stay with current implementation for this part

File: test/unit/uk/gov/hmrc/exports/movements/controllers/ConsolidationControllerSpec.scala [hunk 3]
```diff
@@ -16,325 +16,62 @@
...
-            app,
-            FakeRequest(POST, disassociateDucrUri)
-              .withHeaders(ValidConsolidationRequestHeaders.toSeq: _*)
-              .withJsonBody(exampleDisassociateDucrConsolidationRequestJson)
-          ).get.futureValue
+        val result = controller.submitConsolidation()(postRequest(correctRequest))
 
-          verifyZeroInteractions(submissionServiceMock)
-        }
+        status(result) mustBe ACCEPTED
```
[the remark this answers is not in the document]
Comment (PatrykRudnicki) [reply test/unit/uk/gov/hmrc/exports/movements/controllers/ConsolidationControllerSpec.scala]: Added

File: app/uk/gov/hmrc/exports/movements/models/consolidation/Consolidation.scala [hunk 4]
```diff
@@ -34,51 +29,30 @@ object ConsolidationType extends Enumeration {
...
 
-object ConsolidationRequest {
+object Consolidation {
   implicit val associateDucrFormat = Json.format[AssociateDucrRequest]
   implicit val disassiociateDucrFormat = Json.format[DisassiociateDucrRequest]
   implicit val shutMucrFormat = Json.format[ShutMucrRequest]
 
   implicit val format = Union
-    .from[ConsolidationRequest](typeField = "type")
+    .from[Consolidation](typeField = "type")
```
Comment (ed0906) [inline app/uk/gov/hmrc/exports/movements/models/consolidation/Consolidation.scala]: when this serializes to json you will get a `type` and a `consolidationType` field

I'd change this to `type` and use your enum as the values

File: app/uk/gov/hmrc/exports/movements/models/submissions/ActionType.scala [hunk 5]
```diff
@@ -39,4 +41,10 @@ object ActionType {
       case _                              => JsError("Unknown ActionType")
     }
   }
+
+  def fromConsolidationType(consolidationType: ConsolidationType): ActionType = consolidationType match {
```
Comment (ed0906) [inline app/uk/gov/hmrc/exports/movements/models/submissions/ActionType.scala]: Couldn't this just be an apply method? 

So we can do `ActionType(ConsolidationType.ASSOCIATE_DUCR)` or `ActionType(MovementType.ARRIVE)`

File: app/uk/gov/hmrc/exports/movements/models/consolidation/Consolidation.scala (hunk 4 above)
Comment (PatrykRudnicki) [reply app/uk/gov/hmrc/exports/movements/models/consolidation/Consolidation.scala]: Okay, I'll copy enum to frontend and reuse this

Comment (PatrykRudnicki) [reply app/uk/gov/hmrc/exports/movements/models/consolidation/Consolidation.scala]: Added

File: app/uk/gov/hmrc/exports/movements/models/submissions/ActionType.scala (hunk 5 above)
Comment (PatrykRudnicki) [reply app/uk/gov/hmrc/exports/movements/models/submissions/ActionType.scala]: Good spot, changed

**Comment by PatrykRudnicki** [comment]
test this please

**Pull request merged**